integrations/lib/multiwoven/integrations/destination/zendesk/client.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+module Multiwoven
+  module Integrations
+    module Destination
+      module Zendesk
+        include Multiwoven::Integrations::Core
+        class Client < DestinationConnector
+          prepend Multiwoven::Integrations::Core::RateLimiter
+          def check_connection(connection_config)
+            connection_config = connection_config.with_indifferent_access
+            initialize_client(connection_config)
+            authenticate_client
+            success_status
+          rescue StandardError => e
+            handle_exception("ZENDESK:CHECK_CONNECTION:EXCEPTION", "error", e)
+            failure_status(e)
+          end
+
+          def discover(_connection_config = nil)
+            catalog = build_catalog(load_catalog)
+            catalog.to_multiwoven_message
+          rescue StandardError => e
+            handle_exception("ZENDESK:DISCOVER:EXCEPTION", "error", e)
+            failure_status(e)
+          end
+
+          def write(sync_config, records, action = "create")
+            @action = sync_config.stream.action || action
+            initialize_client(sync_config.destination.connection_specification)
+            process_records(records, sync_config.stream)
+          rescue StandardError => e
+            handle_exception("ZENDESK:WRITE:EXCEPTION", "error", e)
+            failure_status(e)
+          end
+
+          private
+
+          def initialize_client(connection_config)
+            connection_config = connection_config.with_indifferent_access
+            @client = ZendeskAPI::Client.new do |config|
+              config.url = ZENDESK_TICKETING_URL
+              config.username = connection_config[:username]
+              config.password = connection_config[:password]
+            end
+          end
+
+          def authenticate_client
+            response = @client.tickets.page(1).per_page(1).fetch
+
+            raise StandardError, "Failed to authenticate with Zendesk: #{response.errors.join(", ")}" unless response.is_a?(Array) && !response.empty?
```

Comment:
why are we raising an error if the response is empty and not an error? How is this related to authentication? What if the user has zero tickets then will it raise an error?

Reply:
It's related to authentication because it validates that there is a connection to a Zendesk account. The response body should contain a list/array of tickets. The list can be empty (user has zero tickets) so there will be no error raised. However, if the response itself is nil (response is empty), then an error will be raised.

Reply:
@praneeth-ais The purpose of this function is to validate the authentication if we get any other response then a 403 then it is authenticated even if it's null then why do we need this.